include/boost/geometry/algorithms/detail/within/interface.hpp

```diff
@@ -70,13 +70,7 @@ struct within
                              Geometry2 const& geometry2,
```

Comment:
same here. why no dispatching as in area and distance?

Reply:
I answered that before but also here, dispatching based on function templates doesn't force us to explicitly state the types. I've done it there only because I was forced to do it, not because I wanted to do it. ;)